cmd/list.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 ThoughtWorks, Inc.
+
+// This file is part of Gauge.
+
+// Gauge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Gauge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Gauge.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+	supersort "sort"
+
+	"github.com/getgauge/gauge/gauge"
+	"github.com/getgauge/gauge/logger"
+	"github.com/getgauge/gauge/validation"
+	"github.com/spf13/cobra"
+)
+
+var (
+	listCmd = &cobra.Command{
+		Use:     "list [flags] [args]",
+		Short:   "List specifications, scenarios or tags for a gauge project",
+		Long:    `List specifications, scenarios or tags for a gauge project`,
+		Example: `  gauge list --tags specs`,
+		Run: func(cmd *cobra.Command, args []string) {
+			res := validation.ValidateSpecs(getSpecsDir(args), false)
```

Comment:
We don't need to validate all specs here. The `validation.ValidateSpecs` asserts if there are correct implementations in the specific runner. Such errors should not cause this to fail. Also, I don't see the need to start off a language runner here.
We can collect all specs where there are no parse errors and show the respective content.
You can call the parser package's `parser.ParseSpecs` method to get all specifications here.

Reply:
Ah, way better - it is way faster now